MAUI/DataForm/localization.md

```diff
@@ -0,0 +1,183 @@
+
+    ![shows-the-name-of-resource-file-to-be-added-for-maui-dataform](images/localization/shows-the-name-of-resource-file-to-be-added-for-maui-dataform.png)
+
+4. The culture name indicates the name of the language and country.
+
+5. Now, select `Add` option to add the resource file in **Resources** folder.
+
+    ![shows-the-added-resource-file-for-french-language-in-maui-dataform](images/localization/shows-the-added-resource-file-for-french-language-in-maui-dataform.png)
+
+6. Add the Name/Value pair in Resource Designer of `DataFormLocalization.fr.resx` file and change its corresponding value to corresponding culture.
+ 
+    ![shows-the-added-resource-file-name-value-pair-in-the-resource-designer-in-maui-dataform](images/localization/shows-the-added-resource-file-name-value-pair-in-the-resource-designer-in-maui-dataform.png)
+
+## Localizing data form item display values
+
+Here, the display attributes or data form item display values get localized based on culture from Localization Resource File (.Resx).
+
+#### Using attribute
+
+`ResourceType` [Display](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.displayattribute?view=net-7.0) attribute specifies the Resources File (.Resx) which is used to localize the Display attribute of `Name`, `ShortName`, `GroupName`, `Prompt`, and `ItemsSource` values.
+
+{% tabs %}
+{% highlight C# %}
+
+    [Display(Name = "First Name", ShortName = "Name", GroupName = "Details", Prompt = "Type first name", ResourceType = typeof(DataFormLocalization))]
+    public string FirstName { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+You can also localize the DataFormItem `LabelText`, `PlaceHolderText`, `GroupName` in the `GenerateDataFormItem` event of SfDataForm by using the Resources (.Resx) file.
+
+Here, string member of .resx file will be accessed through the class (in resxFilename.Designer.cs) which was auto-generated when .resx file created and static string members get localized using [ResourceManager](https://learn.microsoft.com/en-us/dotnet/api/system.resources.resourcemanager.getstring?view=net-7.0) based on culture.
+
+{% tabs %}
+{% highlight C# %}
+
+[Display(Name = "First Name", GroupName = "Details", Prompt = "Type first name")]
+public String FirstName { get; set; }
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem?.GroupName == "Details")
+        {
+            e.DataFormItem.GroupName = DataFormLocalization.GroupName;
+        }
+        if (e.DataFormItem.LabelText == "First Name")
+        {
+            e.DataFormItem.LabelText = DataFormLocalization.FirstName;
+            e.DataFormItem.PlaceholderText = DataFormLocalization.PlaceHolder;
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Localizing validation error messages
+
```

Comment:
Here, the `ValidationAttributes` or data form error messages get localized based on culture from Localization Resource File (.Resx).

Reply:
content changed